x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,580 @@
+
+**Coverage**: ?.
+
+```Gherkin
+Given user is on the Add Rules page
+When user navigates to the Rule Management page via breadcrumbs
+Then the Rule Management page should be displayed
+```
+
+### Rule upgrade workflow: base cases
+
+#### **Scenario: User can upgrade prebuilt rules one by one**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+When user upgrades one individual rule
+Then success message should be displayed after upgrade
+And the upgraded rule should be removed from the table
+And user should see the number of rules available to upgrade decreased by 1
+```
+
+#### **Scenario: User can upgrade multiple prebuilt rules selected on the page**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+When user selects <Z> rules
+Then user should see a CTA to upgrade <Z> number of rules
+When user clicks the CTA
+Then success message should be displayed after upgrade
+And all the <Z> upgraded rules should be removed from the table
+And user should see the number of rules available to upgrade decreased by <Z> number of upgraded rules
+
+Examples:
+  | Z                               |
+  | a few rules on the page, e.g. 2 |
+  | all rules on the page, e.g. 12  |
+```
+
+#### **Scenario: User can upgrade all available prebuilt rules at once**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+When user upgrades all rules
+Then success message should be displayed after upgrade
+And user should NOT see a CTA to upgrade prebuilt rules
+And user should NOT see a number of rules available to upgrade
+And user should NOT see the Rule Updates table
+```
+
+### Rule upgrade workflow: filtering, sorting, pagination
+
+TODO: add scenarios
+
+### Rule upgrade workflow: misc cases
+
+#### **Scenario: User opening the Rule Updates table sees a loading skeleton until the package installation is completed**
+
+**Coverage**: ?.
+
+```Gherkin
+Given prebuilt rules package is not installed
+When user opens the Rule Management page
+And user opens the Rule Updates table
+Then user should see a loading skeleton until the package installation is completed
+```
```

Comment:
TODO: @banderror sync with @jpdjere on coverage

Reply:
Synced and removed -- the tab won't be shown until the package is installed. Replaced by this scenario:

#### **Scenario: User don't see the Rule Updates tab until the package installation is completed**

**Automation**: unit tests.

```Gherkin
Given prebuilt rules package is not installed
When user opens the Rule Management page
Then user should NOT see the Rule Updates tab until the package installation is completed and there are rules available for upgrade
```